### PR TITLE
[neon/acr] impr: push to ACR while building images

### PR DESCRIPTION
**Pull request by antonyc**
Merge this strictly after neoneastus2 terraformed.

This tests the ability to push into ACR using OIDC. Previos run in this branch [proves that](https://github.com/neondatabase/neon/actions/runs/10149694500/job/28065170110#step:7:6). In `promote-images` we push the following images `neon compute-tools {vm-,}compute-node-{v14,v15,v16}` into neoneastus2.

https://github.com/neondatabase/cloud/issues/14640

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -859,6 +859,28 @@ jobs:
...
+        run: |
+          az acr login --name=neoneastus2
+
+      - name: Copy docker images to ACR-dev
+        if: github.ref_name == 'main'
+        run: |
+          for image in neon compute-tools {vm-,}compute-node-{v14,v15,v16}; do
+            docker buildx imagetools create -t \
+              neoneastus2.azurecr.io/neondatabase/${image}-${version}:${{ needs.tag.outputs.build-tag }} \
+                                     neondatabase/${image}-${version}:${{ needs.tag.outputs.build-tag }}
```
Comment (bayandin) [inline .github/workflows/build_and_test.yml:881]: - We don't need `-${version}` part here (it's already in the `${image}`)
- I'd split lines a bit differently and would keep `-t` on the same line as the new image (`docker buildx imagetools create` accepts several `-t` options and several arguments)

```suggestion
            docker buildx imagetools create \
              -t neoneastus2.azurecr.io/neondatabase/${image}:${{ needs.tag.outputs.build-tag }} \
                                       neondatabase/${image}:${{ needs.tag.outputs.build-tag }}
```

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -859,6 +859,28 @@ jobs:
                                                neondatabase/vm-compute-node-${version}:${{ needs.tag.outputs.build-tag }}
           done
 
+      - name: Azure login
+        if: github.ref_name == 'main'
+        uses: azure/login@6c251865b4e6290e7b78be643ea2d005bc51f69a  # @v2.1.1
+        with:
+          client-id: 4bb00e03-987d-4b86-a013-079729a85cb8
+          tenant-id: c8350122-1697-4543-929a-d4a75d1bb552
+          subscription-id: a57dcb0f-4249-4982-a872-1b73a7126c12
```
Comment (bayandin) [inline .github/workflows/build_and_test.yml:868]: Just in case, is it ok to share these values in a public repo?

**Review by bayandin — COMMENTED** [review]
LGTM

Before merging it, let's remove `if: github.ref_name == 'main'` part and see if it work in the PR

File: .github/workflows/build_and_test.yml (hunk 2 above)
Comment (fcdm) [reply .github/workflows/build_and_test.yml:868]: They are not secrets but I think we still shouldn't leak them.

**Comment by antonyc** [comment]
> Before merging it, let's remove if: github.ref_name == 'main' part and see if it work in the PR

I had it pushed in previous version https://github.com/neondatabase/neon/actions/runs/10149694500/job/28065170110#step:7:6

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (antonyc) [reply .github/workflows/build_and_test.yml:881]: > We don't need -${version} part here (it's already in the ${image})

Ah, thanks!

> I'd split lines a bit differently and would keep -t on the same line 

Sure.

**Review by fcdm — APPROVED** [review]

**Comment by antonyc** [comment]
Checked that it worked.

**Pull request merged**